frontend/packages/kubevirt-plugin/src/components/vms/vm-details.tsx

```diff
@@ -29,8 +29,9 @@ export const VMDetailsFirehose: React.FC<VMTabProps> = ({
   templates,
   customData: { kindObj },
 }) => {
-  const vm = kindObj === VirtualMachineModel && isVM(objProp) ? objProp : vmProp;
-  const vmi = kindObj === VirtualMachineInstanceModel && isVMI(objProp) ? objProp : vmiProp;
+  const vm = kindObj === VirtualMachineModel && isVM(objProp) ? objProp : isVM(vmProp) && vmProp;
```

Comment:
how come? The types in VMTabProps should always corespond no? How does this help?

Are the objects different when you look at them in debug, and why are they different?

Reply:
> how come? The types in VMTabProps should always corespond no?
> Are the objects different when you look at them in debug, and why are they different?

If we do not find an object with matching type. `vmProp` is an empty object when no matching vm is found, since we check `if (vm) ...`  we get a false `true` becase `{}` is true.

> How does this help?

So we need to check `vmProp` (and `vmiProp`) for empty objects.

Reply:
ah I see, it is that weird behaviour when firehose gives empty object. Could we at least pass `null` instead of `false` when that happens?

Reply:
sure :+1: done